src/subdomains/generic/user/models/user/user.service.ts

```diff
@@ -240,21 +245,44 @@ export class UserService {
     return this.userRepo.save({ ...user, ...update });
   }
 
-  async deactivateUser(id: number, allUser = false): Promise<void> {
-    const mainUser = await this.userRepo.findOne({
-      where: { id },
-      relations: { userData: { users: true, kycSteps: true } },
+  async updateAddress(userDataId: number, address: string, dto: UpdateAddressDto): Promise<UserV2Dto> {
+    const userData = await this.userDataRepo.findOne({
+      where: { id: userDataId },
+      relations: { users: { wallet: true } },
     });
-    if (!mainUser) throw new NotFoundException('User not found');
-    if (mainUser.userData.isBlockedOrDeactivated) throw new BadRequestException('User Account already deactivated');
-    if (mainUser.isBlockedOrDeactivated) throw new BadRequestException('User already deactivated');
+    if (!userData) throw new NotFoundException('User not found');
+
+    const userToUpdate = userData.users.find((u) => u.address === address);
+    if (!userToUpdate) throw new NotFoundException('Address not found');
+
+    userToUpdate.label = dto.label;
+    await this.userRepo.save(userToUpdate);
```

Comment:
Would be better to use the `update` method.
I would make an `setLabel` factory method in the user entity. Similar to `activateUser`. Just name it setLabel and return `UpdateResult<User>`, same as in `activateUser` and then you can easy use `await this.userRepo.update(...userToUpdate.setLabel(dto.label))`